docs/proposals/service-catalog.md

```diff
@@ -0,0 +1,408 @@
+        are required to use the system so that my users can easily consume
+        them in their own namespaces
+    2.  As a user of a software system running in Kubernetes, I want to consume
+        the shared resources associated with that system in my own namespace so
+        that I can use the system in my application
+4.  Sharing unique resources for a service:
+    1.  As an operator of a software system, I want to be able to generate a
+        unique resource for each user that wants to use the system so that I can
+        manage permissions granularly
+    2.  As a user of a software system, I want to get a unique set of resources
+        for system in my namespace so that I can use the system in my
+        application
+5.  Policy for viewing and using services
+    1.  As a service operator, I want to be able to describe a policy for who is
+        able to view my service, so that I can ensure that only the right users
+        have access to see that my service exists
+    2.  As a service operator, I want to be able to describe a policy for who is
+        able to use my service, so that I can ensure that users have the right
+        degree of autonomy when using my service
+6.  Consuming services - visibility
+    1.  As a service operator, I want to be able to see the users consuming my
+        service and track their usage of my service, so that I can be aware of
+        the consumers of the service and charge them according to their usage
+    2.  As an developer consuming services, I want to be able to see the
+        services are being consumed, so that I can ensure that I am consuming
+        only services that I need
+7.  Resource provisioning
+    1.  As a service operator, I want to be able to provision new resources for
+        a user when they begin to consume my service so that the user will have
+        their own resources to use for my service
+
+### Advertising services and recipes
+
+Within and without a Kubernetes cluster, there are services that users wish to
+highlight and make available to other users.  Users might also wish to publish
+recipes that allow other users to run their own services.  Some examples:
+
+1.  A user's namespace contains `etcd`, `etcd- discovery`, and `postgresql`
+    services, and the only one the user wants to share with others is the
+    `postgresql` service
+2.  A SaaS product like a externally hosted database for which a Kubernetes
+    Service exists to provide a stable network endpoint
+3.  A user makes a database run in Kubernetes and wants to share their recipe
+
+In order to share these services, there has to be a central place where they can
+be registered and advertised.  In this proposal, we'll call this place the
+'catalog'.
+
+### Labeling catalog entries
+
+A user making an advertisement in the catalog should be able to label their
+offerings.  Labeling an entry in the catalog lets users search by category or
```

Comment:
Thinking on k-v labels vs. value-only tags, I think labels would be useful. A formal or informal taxonomy could be developed analogous to [ID3 attributes](https://en.wikipedia.org/wiki/ID3#Layout)